### ISSUE TITLE
Release rejected when multiple formats in torrent

### ISSUE THREAD

**Issue by SoCraDi** [Readarr/Readarr#1440]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

EPUB is set as the only format for the ebook profile. Readarr is ignoring a torrent that contains multiple formats (AZWR3, EPUB, MOBI) but it should not have been rejected as it contains EPUB. 

Example release title/name: Pride and Prejudice by Jane Austen [ENG / azw3 epub mobi]
Screenshots: https://imgur.com/a/Vi2WUt0

In the example above, EPUB is in the release title/name and release should not have been rejected. I am not sure how Readarr determines what file formats there are in the release name but it could be parsing the name and AZW3 is listed first and it may stop parsing as soon as it finds the first ebook format in the string.

Relevant trace logs:
|Parser|Quality parsed: AZW3 v1
|QualityAllowedByProfileSpecification|Checking if report meets quality requirements. AZW3 v1
|Debug|QualityAllowedByProfileSpecification|Quality AZW3 v1 rejected by Author's quality profile
|Debug|DownloadDecisionMaker|Release rejected for the following reasons: [Permanent] AZW3 is not wanted in profile

### Expected Behavior

Readarr should not have rejected the release

### Steps To Reproduce

1. Quality profile on the author = Ebook (EPUB is the only format, upgrades allowed, upgrade until EPUB)
2. Add Jane Austen as an author
3. Find book "Pride and Prejudice" in the list, execute manual search. Ensure MaM is added as an indexer
4. Find release "Pride and Prejudice by Jane Austen [ENG / azw3 epub mobi]" in the list which is identified as "AZW3" and is rejected even though it contains EPUB. Screenshots: https://imgur.com/a/Vi2WUt0

### Environment

```markdown
- OS: Synology DSM7
- Readarr: 0.1.0.1154
- Docker Install: yes
- Using Reverse Proxy: no
- Browser: N/A
```

### What branch are you running?

Nightly

### Trace Logs?

https://gist.github.com/SoCraDi/507b18b625014261e71673e52a07d975

[AB#2299](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2299)

**Comment by wietsedv** [comment]
Is there a workaround for this problem? Readarr also does not postprocess the epub after manual download, because it thinks it's a azw3 book, even though there is an epub in the download directory.

**Comment by bakerboy448** [comment]
> Is there a workaround for this problem? Readarr also does not postprocess the epub after manual download, because it thinks it's a azw3 book, even though there is an epub in the download directory.

Manual Import in Readarr.

No that does not mean manually copying any file.

**Comment by nonamethanks** [comment]
This issue pretty much breaks getting files from myanonamouse because they prefer multi-format packs to single-file torrents.

**Comment by bakerboy448** [comment]
> This issue pretty much breaks getting files from myanonamouse because they prefer multi-format packs to single-file torrents.

Easily worked around by wanting the common 3 file types

**Comment by trentks** [comment]
Sorry for the duplication!

Just thought I'd mention that azw3 isn't supported by kobo, and because I think it's conventional on MaM for multi-format releases to be tagged in alphabetical order, they all get incorrectly flagged as azw3.  It's not really a work around to "want the common 3 file types", you'll just wind up with a boat load of azw3s in your library which are a nuisance format.

**Comment by DireMunchkin** [comment]
@trentks There is a better workaround than that I've found - Which is that you just manually download the multiformat release for the eBook you want. The importer handles it correctly so i.e. if you only want EPUB format and the release has a EPUB a AZW3 and a MOBI file in a folder, only the EPUB will be imported. The problem is only that Readarr will not automatically download a multiformat release to get a EPUB.

**Comment by trentks** [comment]
> @trentks There is a better workaround than that I've found - Which is that you just manually download the multiformat release for the eBook you want. The importer handles it correctly so i.e. if you only want EPUB format and the release has a EPUB a AZW3 and a MOBI file in a folder, only the EPUB will be imported. The problem is only that Readarr will not automatically download a multiformat release to get a EPUB.

Yeah, that's my current way around it... but as you've said lacking the ability to auto grab is a MASSIVE drawback. The way it currently exists you have to choose between having a hoard of azw3s hanging around in your library, or you have to go full hands-on. Neither are great.
